src/Microsoft.Net.Http.Headers/NameValueHeaderValue.cs

```diff
@@ -142,6 +147,20 @@ public override bool Equals(object obj)
             }
         }
 
+        public StringSegment GetDecodedValue()
+        {
+            return HeaderUtilities.RemoveQuotes(_value);
+        }
+
+        public void SetAndEncodeValue(StringSegment value)
+        {
+            if (_quotedValue)
+            {
+                value = $"\"{value}\"";
```

Comment:
1. Why check `_quotedValue`? That's about the previous `Value`.
2. This doesn't "encode" anything, especially double-quote characters _within_ the `value`.